cachi2/core/package_managers/yarn/resolver.py

```diff
@@ -344,13 +342,40 @@ def _generate_purl_for_package(package: _ResolvedPackage, project: Project) -> s
     :param project: the project object to resolve the configured registry url and file paths
         for file dependencies.
     """
-    # registry url can be accessed in project.yarnrc
-    # paths for file dependencies are relative to project.source_dir
+    qualifiers = dict()
+    if package.checksum is not None:
+        qualifiers["checksum"] = package.checksum
```

Comment:
Another detail we need to handle here is to prefix the checksum with the algorithm used (I'm not sure what it is, we need to look into it). The only place I could find that mentions it in the spec is [here](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#known-qualifiers-keyvalue-pairs), but there might be more info there.

Reply:
[sha512](https://github.com/yarnpkg/berry/blob/017b94ae4eb20dea14ac673a053a1f2974b778ff/packages/yarnpkg-core/sources/hashUtils.ts#L84)

Reply:
The other piece of useful information is that they actually use that function for checksums within the lockfile: https://github.com/yarnpkg/berry/blob/017b94ae4eb20dea14ac673a053a1f2974b778ff/packages/yarnpkg-core/sources/Project.ts#L207.
IMO we need to keep references to this information in a code commentary for future reference. This may come handy at some point since the hash algorithm used for checksums hasn't been mentioned anywhere in Yarn docs and thus the algorithm used can be changed by Yarn at will without us even noticing (e.g. when sha512 becomes insecure) ...unless we want to start recomputing checksums  just to make sure the algorithm still checks.